netket/optimizer/stochastic_reconfiguration.py

```diff
@@ -0,0 +1,303 @@
+import numpy as _np
+from scipy.linalg import lstsq as _lstsq
+from scipy.sparse.linalg import LinearOperator
+from netket.stats import sum_on_nodes as _sum_on_nodes
+from scipy.sparse.linalg import cg, gmres, minres
+from mpi4py import MPI
+
+
+class SR:
+    r"""
+    Performs stochastic reconfiguration (SR) updates.
+    """
+
+    def __init__(
+        self,
+        lsq_solver=None,
+        diag_shift=0.01,
+        use_iterative=True,
+        is_holomorphic=True,
+        svd_threshold=None,
+        sparse_tol=None,
+        sparse_maxiter=None,
+    ):
+
+        self._lsq_solver = lsq_solver
+        self._diag_shift = diag_shift
+        self._use_iterative = use_iterative
+        self._is_holomorphic = is_holomorphic
+        self._svd_threshold = svd_threshold
+        self._scale_invariant_pc = False
+        self._S = None
+        self._last_rank = None
+
+        # Temporary arrays
+        self._v_tilde = None
+        self._res_t = None
+
+        # Quantities for sparse solver
+        self.sparse_tol = 1.0e-5 if sparse_tol is None else sparse_tol
+        self.sparse_maxiter = sparse_maxiter
+        self._lsq_solver = lsq_solver
+        self._x0 = None
+        self._make_solver()
+
+        self._comm = MPI.COMM_WORLD
+
+    def _make_solver(self):
```

Comment:
Maybe call this `_init_solver`. `make_...` sounds like the method should return something, and not modify state of the object.